### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by atrzaska**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    working_directory: ~/app
+    steps:
+      - checkout
+      - run: mix local.hex --force
+      - run: mix local.rebar --force
+      - run: mix deps.get
+      - run: mix compile --warnings-as-errors
+      - run: dockerize -wait tcp://localhost:5432 -timeout 1m
+      - run: mix ecto.migrate
+      - run: mix test
```
Comment (uesteibar) [inline .circleci/config.yml:29]: we can also add `mix dialyzer` and `mix credo` here

Comment (atrzaska) [reply .circleci/config.yml:29]: added

Comment (atrzaska) [reply .circleci/config.yml:29]: dialyzer is not found

Comment (atrzaska) [reply .circleci/config.yml:29]: do you have idea why it is failing ? https://app.circleci.com/pipelines/github/retro-tool/retro-tool-api/10/workflows/6d8eab7e-ddab-40fc-a593-cc6322faa34b/jobs/10

Locally dialyzer works

Comment (uesteibar) [reply .circleci/config.yml:29]: Got it! I'd say we need to add it also for the test environment here: https://github.com/retro-tool/retro-tool-api/blob/master/mix.exs#L72

Comment (atrzaska) [reply .circleci/config.yml:29]: Let's see if that fixes the problem

Comment (atrzaska) [reply .circleci/config.yml:29]: That fixed the problem

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,7 +21,8 @@ jobs:
       - run: mix local.hex --force
       - run: mix local.rebar --force
       - run: mix deps.get
-      - run: mix compile --warnings-as-errors
+      - run: mix compile.elixir --warnings-as-errors
+      - run: mix compile
```
Comment (uesteibar) [inline .circleci/config.yml]: what's the reason for this change?

Comment (atrzaska) [reply .circleci/config.yml]: warnings-as-errors is a flag on `mix compile.elixir` not `mix compile` 


reference: https://hexdocs.pm/mix/Mix.Tasks.Compile.Elixir.html
https://hexdocs.pm/mix/Mix.Tasks.Compile.html#summary

Comment (atrzaska) [reply .circleci/config.yml]: btw we have some warnings so build is failing

Comment (atrzaska) [reply .circleci/config.yml]: I'll fix it on the long weekend

Comment (uesteibar) [reply .circleci/config.yml]: oh no, I promise it's on `mix compile` too, I run it every day

Comment (atrzaska) [reply .circleci/config.yml]: ok, reverted

**Comment by atrzaska** [comment]
We have 2 errors remaining from dialyzer 

```
Total errors: 2, Skipped: 0, Unnecessary Skips: 0
 
done in 0m2.45s
 
:0:unknown_function
 
Function ExUnit.Callbacks.__merge__/3 does not exist.
 
________________________________________________________________________________
 
:0:unknown_function
 
Function ExUnit.CaseTemplate.__proxy__/2 does not exist.
 
________________________________________________________________________________
 
```

I can check them later

**Comment by uesteibar** [comment]
@atrzaska as for those errors, I see we're running dialzyer with `MIX_ENV=test` so it runs it also for test files. We should only run it for our compiled files though, so `MIX_ENV=dev mix dialyzer` should make this green 👍
